### PR TITLE
 Distillation: Log per-step TFLOPs (student fwd+bwd, teacher fwd-only)

### PR DESCRIPTION
**Pull request by gagika**
# Description

- Add `calculate_distillation_tflops_per_device` and wire it into `MaxTextDistillationTrainer` to log
  per-step per-device TFLOPs to TensorBoard (`perf/per_device_tflops{,_student,_teacher}` and `_per_sec` 
  variants).
- Console summary per step: `step N | step_time=Xs | TFLOPs/s/device: Y (student=A, teacher=B)`.       

- Replace the raw `Distillation metrics: {dict}` dump in `_post_process_train_step` with a compact,    
  selective line; optional metrics (feature loss, MoE LB) hide when zero.                                

# Tests

Unit test and end to end test checking the logs.

# Checklist

Before submitting this PR, please make sure (put X in square brackets):
- [x] I have performed a self-review of my code. For an optional AI review, add the `gemini-review` label.
- [x] I have necessary comments in my code, particularly in hard-to-understand areas.
- [x] I have run end-to-end tests tests and provided workload links above if applicable.
- [x] I have made or will make corresponding changes to the doc if needed, including adding new documentation pages to the relevant Table of Contents (toctree directive) as explained in [our documentation](https://maxtext.readthedocs.io/en/latest/development.html#adding-new-documentation-files).

### CODE REVIEW

File: src/maxtext/trainers/post_train/distillation/distillation_utils.py [hunk 1]
```diff
@@ -251,6 +252,35 @@ def weighted_mean(sum_count_pairs: Sequence[tuple[Any, Any]] | np.ndarray) -> fl
...
+
+  Returns:
+    (combined, student, teacher) per-step per-device TFLOPs.
+  """
+  student_tflops, _, _ = maxtext_utils.calculate_tflops_training_per_device(student_config, log=False)
+  if is_offline or teacher_config is None:
+    teacher_tflops = 0.0
+  else:
+    teacher_full, _, _ = maxtext_utils.calculate_tflops_training_per_device(teacher_config, log=False)
+    teacher_tflops = teacher_full / 3.0
```
Comment (vlad-karp) [inline src/maxtext/trainers/post_train/distillation/distillation_utils.py:295]: what is 3.0 ?

File: src/maxtext/trainers/post_train/distillation/train_distill.py [hunk 2]
```diff
@@ -376,7 +419,39 @@ def _post_process_train_step(self, aux: dict[str, tuple[jax.Array, jax.Array]])
...
+
+    # Compact per-step summary: only the metrics that change run-to-run, and
+    # only those that are nonzero (so MoE / feature-distill terms hide when off).
+    # `aux` values are (sum, count) tuples; reduce to scalar via sum/count.
+    def _scalar(v):
+      s, c = v
+      c = float(c)
+      return float(s) / c if c > 0 else 0.0
+
+    headline_keys = (
```
Comment (vlad-karp) [inline src/maxtext/trainers/post_train/distillation/train_distill.py:431]: we use these string constants in the loss compute. Maybe create static variables for them?

File: src/maxtext/trainers/post_train/distillation/distillation_utils.py (hunk 1 above)
Comment (gagika) [reply src/maxtext/trainers/post_train/distillation/distillation_utils.py:295]: calculate_tflops_training_per_device compute training flops which is fwd + bwd = 3 * fwd pass (bwd = 2 fwd), I have a description at the function description, let me know if you prefer a comment here too.

File: src/maxtext/trainers/post_train/distillation/train_distill.py (hunk 2 above)
Comment (gagika) [reply src/maxtext/trainers/post_train/distillation/train_distill.py:431]: done, ptl

**Review by JamesDeng42 — APPROVED** [review]
LGTM

**Review by vlad-karp — APPROVED** [review]

File: src/maxtext/trainers/post_train/distillation/distillation_utils.py (hunk 1 above)
Comment (vlad-karp) [reply src/maxtext/trainers/post_train/distillation/distillation_utils.py:295]: ok, but this is just a rough approximation, right? 
Also, if remat is enabled, it may be a different factor, right?

Comment (gagika) [reply src/maxtext/trainers/post_train/distillation/distillation_utils.py:295]: remat only impacts student model (not teacher), and it won't impact Model Flops (it impacts hardware flops).
`calculate_tflops_training_per_device` internally uses bwd = 2*fwd it's quite accurate AFAIK.

If we have some frozen weights, then calculate_tflops_training_per_device might overestimate the model flops.
